### PR TITLE
feat: generalize e2e test for archway 

### PR DESCRIPTION
**Pull request by debendraoli**
### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: test/e2e/config-icon-icon.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      name: goloop1
+      chain_id: icon1
+      bech32_prefix: hx
+      bin: "/goloop/bin/goloop"
+      image:
+        repository: goloop/gochain-icon
+        version: latest
+        uid_gid: ''
+    contracts:
+      ibc: "$BASE_PATH/contracts/javascore/gochain-btp/contracts/ibc-optimized.jar"
```
Comment (AntonAndell) [inline test/e2e/config-icon-icon.yaml:20]:
```suggestion
      ibc: "$BASE_PATH/contracts/javascore/ibc/build/libs/ibc-0.1.0-optimized.jar"
```

File: test/e2e/config-icon-icon.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      name: goloop2
+      chain_id: icon2
+      bech32_prefix: hx
+      bin: "/goloop/bin/goloop"
+      image:
+        repository: goloop/gochain-icon
+        version: latest
+        uid_gid: ''
+    contracts:
+      ibc: "$BASE_PATH/contracts/javascore/gochain-btp/contracts/ibc-optimized.jar"
```
Comment (AntonAndell) [inline test/e2e/config-icon-icon.yaml:43]:
```suggestion
      ibc: "$BASE_PATH/contracts/javascore/ibc/build/libs/ibc-0.1.0-optimized.jar"
```

File: test/e2e/config.yaml [hunk 3]
```diff
@@ -1,47 +1,53 @@
...
+      name: goloop1
+      chain_id: icon1
+      bech32_prefix: hx
+      bin: "/goloop/bin/goloop"
+      image:
+        repository: goloop/gochain-icon
+        version: latest
+        uid_gid: ''
+    contracts:
+      ibc: "$BASE_PATH/contracts/javascore/gochain-btp/contracts/ibc-optimized.jar"
```
Comment (AntonAndell) [inline test/e2e/config.yaml:20]:
```suggestion
      ibc: "$BASE_PATH/contracts/javascore/ibc/build/libs/ibc-0.1.0-optimized.jar"
```

File: test/chains/icon/localnet.go [hunk 4]
```diff
@@ -327,7 +329,7 @@ func (c *IconLocalnet) SetupIBC(ctx context.Context, keyName string) (context.Co
 		return nil, err
 	}
 
-	client, err := c.getFullNode().DeployContract(ctx, c.scorePaths["client"], c.keystorePath, "ibcHandler="+ibcAddress)
```
Comment (AntonAndell) [inline test/chains/icon/localnet.go:330]: This can be reverted before we commit to main. It was needed when testing with a old client contract

**Review by AntonAndell — CHANGES_REQUESTED** [review]
Some more minor comments, then ready to merge

**Review by AntonAndell — APPROVED** [review]
Looks good to merge

**Review by redlarva — APPROVED** [review]
looks good

**Pull request merged**